### ISSUE TITLE
Handling of Stripe JS with GDPR & minifier plugins

### ISSUE THREAD

**Issue by frosso** [Automattic/woocommerce-payments#9709]
### Describe the bug

The checkout button is missing in mobile view on the WooCommerce site when using WooPayments with the Cookiebot plugin enabled. The issue appears to stem from the Cookiebot plugin altering the loading behavior of the Stripe JavaScript. Specifically, Cookiebot changes the `<script>` type of Stripe’s script from `text/javascript` to `text/plain` due to cookie consent settings, which prevents Stripe from loading correctly, thereby causing WooPayments elements not to display on checkout.

### To Reproduce

1. Enable the Cookiebot plugin (for example)
2.  Go to the WooCommerce checkout page
1. Check if the WooPayments checkout button and Stripe elements are visible.
1. Open the browser console to observe errors.

### Actual behavior

In mobile view, the checkout button is missing on the WooCommerce checkout page. The console displays the following error:

```js
blocks-checkout.js?ver=8.2.2:2 Uncaught (in promise) Error: Invalid prop `stripe` supplied to `Elements`. We recommend using the `loadStripe` utility from `@stripe/stripe-js`.
```

Additionally, attempting to manually initialize Stripe in the console returns:

```js
ReferenceError: Stripe is not defined
```

Inspecting the page source reveals that the Stripe script is enqueued with `type="text/plain"` instead of `type="text/javascript"`, likely due to Cookiebot’s handling of cookie consent.
https://plugins.trac.wordpress.org/browser/cookiebot/trunk/src/lib/helper.php#L151

### Expected behavior

The WooPayments checkout button and Stripe payment fields should load and be fully functional on the checkout page in mobile view, without requiring additional manual actions. The Stripe JavaScript should load without interference from Cookiebot

### Additional context

Potential solution ideas include leveraging [JavaScript Proxies](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) to detect when `Stripe` is added to the `window` object and then triggering the initialization of WooPayments.

p1727770367173279-slack-CGGCLBN58
p1722007079535099-slack-CU6SYV31A

**Comment by danielmx-dev** [comment]
Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Heisenberg-63586f2ec60359001310fa6a/issues/Automattic/woocommerce-payments/9709?planning-poker) with Zenhub @FangedParakeet

**Comment by danielmx-dev** [comment]
@frosso I have followed the instructions but have some issues replicating the issues, could you please let me know if there's a specific setting you are using?

I'm using the following Cookiebot settings:

| Layout | Design | Compliance |
|--------|--------|--------|
|![Image](https://github.com/user-attachments/assets/e61a113a-a9ef-4a2a-be0b-1a4b6ab62032) |![Image](https://github.com/user-attachments/assets/a223705e-ac82-44af-883b-a91e1226879a) | ![Image](https://github.com/user-attachments/assets/1d08d17c-dd55-43cb-906f-0cb9d519411c) | 

Settings from WP-Admin:

| General | Additional |
|--------|--------|
|![Image](https://github.com/user-attachments/assets/616653ac-709a-483e-9506-fa04f06d206e) |![Image](https://github.com/user-attachments/assets/475dbd6a-f657-4ef7-89cf-80233d976ae2) | 

Everything else is set to the default values (I'm not using Google Tag Manager or anything else).

When I load the page I don't see any errors in the console, I'm using the dev tools screen emulator (I'm not sure if that's what you meant as "mobile view"?):

![Image](https://github.com/user-attachments/assets/134a96b1-f1a9-4604-8e83-6d126fb0e652)

**Comment by danielmx-dev** [comment]
It looks like Cookiebot provides an [Automatic Cookie Blocking](https://support.cookiebot.com/hc/en-us/articles/360009074960-Automatic-cookie-blocking) feature that will scan your site and determine which cookies and scripts will be blocked.

> If a script has been identified as the initiator of several cookies and these cookies belong to different categories, then this script will be blocked unless all categories that the cookies belong to have been consented to.

The scanner most likely wasn't able to scan any references to Stripe in my test site, so when I reached checkout it was never blocked.

Cookiebot also provides a feature to manually block some scripts, but I don't think it's possible at all to block the Stripe script this way (unless you force it as described below).

In order to force the issue in my testing environment, I had to edit the plugin code and do the following changes:

* Use "Manual cookie-blocking mode" in the Cookiebot settings (WP Admin).
* Update `Script_Loader_Tag::__construct` to always set the `script_loader_tag` filter.
* Add the following line to the end of `Cookiebot_Addons::build_container` 
```
$this->container->get( 'Script_Loader_Tag_Interface' )->add_tag( 'stripe', array( 'marketing', 'statistics' ) );
```

Now when I open checkout I can see the error. I'll move forward using this method.
